File: agora-easycall-ios-oc/AgoraSample/Manager/AGSAgoraKitManager.m
//
//  AGSAgoraKitManager.m
//  AgoraSample
//
//  Created by apple on 15/9/17.
//  Copyright (c) 2015年 Agora. All rights reserved.
//

#import "AGSAgoraKitManager.h"

static AGSAgoraKitManager *obj = nil;

@implementation AGSAgoraKitManager

+ (AGSAgoraKitManager *)shareManager
{
    static dispatch_once_t once_token;
    dispatch_once(&once_token, ^{
        if (!obj) {
            obj = [[AGSAgoraKitManager alloc] init];
        }
    });
    return obj;
}

- (id)init
{
    self = [super init];
    if (self) {
        self.callBackView = [[AGSCallBackView alloc]init];
        self.callBackView.delegate = self;
    }
    
    return self;
}

- (void)setVendorKey:(NSString *)vendorKey
{
    _vendorKey = vendorKey;
    if (!self.agoraKit) {
        self.agoraKit = [AgoraRtcEngineKit sharedEngineWithVendorKey:vendorKey error:^(AgoraRtcErrorCode errorCode) {
            _errorCode = errorCode;
        }];
    }
}

- (void)didHideCallBackView:(AGSCallBackView *)view
{
    if ([self.delegate respondsToSelector:@selector(didBackChatViewAndHideCallBackView:)]) {
        [self.delegate didBackChatViewAndHideCallBackView:view];
    }
}

@end

File: agora-easycall-ios-swift/SampleSwift/AGSAgoraKitManager.swift
//
//  AGSAgoraKitManager.swift
//  SampleSwift
//
//  Created by let on 10/30/15.
//  Copyright © 2015 Agora. All rights reserved.
//

import UIKit

class AGSAgoraKitManager: NSObject, AGSCallBackViewDelegate {
    
    static let shareManager = AGSAgoraKitManager()
    
    var agoraKit: AgoraRtcEngineKit!
    var vendorKey: String!
    var errorCode: AgoraRtcErrorCode?
    var callBackView: AGSCallBackView!
    var delegate: AGSAgoraKitManagerDelegate?
    
    override init() {
        super.init()
        
        callBackView = AGSCallBackView()
        callBackView.delegate = self
    }
    
    func didHideCallBackView(view: AGSCallBackView) {
        delegate?.didBackChatViewAndHideCallBackView(view)
    }
    
    
}

protocol AGSAgoraKitManagerDelegate {
    func didBackChatViewAndHideCallBackView(view: AGSCallBackView)
}